src/Npgsql/Internal/TypeHandlers/SystemTextJsonHandler.cs

```diff
@@ -27,7 +25,7 @@ namespace Npgsql.Internal.TypeHandlers;
 /// should be considered somewhat unstable, and may change in breaking ways, including in non-major releases.
 /// Use it at your own risk.
 /// </remarks>
-public class JsonHandler : NpgsqlTypeHandler<string>, ITextReaderHandler
+public class SystemTextJsonHandler : NpgsqlTypeHandler<string>, ITextReaderHandler
```

Comment:
Wouldn't it be better to inherit from `JsonTextHandler`? That way while overriding we can just handle the thing we need first, and after that return the result of base implementation?

Reply:
I tried :)

The main problem is that SystemTextJsonHandler (and JsonNetJsonHandler) support arbitrary POCOs, so they can't first check "is this something I specifically support" (because they support anything). The alternative is to expose something like TryValidateAndGetLength from JsonTextHandler and use that for an up-front check; but Try* methods don't work well with async (because out vars aren't supported). It's still possible to expose something like IsTypeSupported from JsonTextHandler, but the value seemed a bit low...

Let me know what you think.

Reply:
Also, might be a good idea to put `typeof(TAny) == typeof(string)` checks behind a method in the base class (we already miss `typeof(TAny) == typeof(ReadOnlyMemory<byte>)` check for `JsonNetJsonHandler`).

Reply:
OK, I guess we can go with IsTypeSupported on the base class, I'll do that.